### ISSUE TITLE
FromSQL: Provide a method to inline parameter in interpolated string

### ISSUE THREAD

**Issue by jeremycook** [dotnet/efcore#14543]
The following code errors with:

> No mapping to a relational type can be found for the CLR type 'RawSqlString'.

```c#
var qualifiedTable = new RawSqlString($"{entry.Metadata.Relational().Schema}.{entry.Metadata.Relational().TableName}");
var table = new RawSqlString(entry.Metadata.Relational().TableName);
await db.Database.ExecuteSqlCommandAsync($"update {table} set Discriminator = {newDiscriminator} from {qualifiedTable} where Id = {entity.Id}");
```
I would like to see the contents of `RawSqlString` or some new type be treated as raw SQL, and not parameterized in anyway. Something analogous to `@Html.Raw("<br>")` but in the context of an interpolated string of SQL.

Currently, I can do something like the following but that results in the EF1000 "... embeds data that will not be parameterized" warning. In general that warning is extremely helpful, even in the example given above.

```c#
string sql = $"update {entry.Metadata.Relational().TableName} set Discriminator = @p0 from {entry.Metadata.Relational().Schema}.{entry.Metadata.Relational().TableName} where Id = @p1";
await db.Database.ExecuteSqlCommandAsync(sql, newDiscriminator, entity.Id);
```

Is there a way to opt-into passing in raw SQL strings that will not be parameterized or interpreted in anyway, and avoids the EF1000 "... embeds data that will not be parameterized" warning? The use case is executing SQL with table and column names determined at runtime that also take advantage of string interpolation of user provided data.

**Comment by smitpatel** [comment]
> Is there a way to opt-into passing in raw SQL strings that will not be parameterized or interpreted in anyway, and avoids the EF1000 "... embeds data that will not be parameterized" warning? 

Yes, disable the warning in source code when using it. Analyzer is warning you that it will not be parameterized and that's what you want so you disable the warning.

Meanwhile, `RawSqlString` is not the type which will be treated as raw SQL.

**Comment by jeremycook** [comment]
@smitpatel I guess I should have phrased this as a feature request then. Yes, I know how to disable those warning, and that RawSqlString is not interpreted that way. My request is for it or something equivalent to `RawSqlString` to be interpretted that way. I've talked about it here https://github.com/aspnet/EntityFrameworkCore/issues/10996#issuecomment-438729944 but it didn't seem like the best place so I opened this issue with some specific error and warning messages to further the discussion.

Can this be reopened as a feature request? I'd be happy to change the title to something like "Support raw SQL in interpolated strings" if that would help.
